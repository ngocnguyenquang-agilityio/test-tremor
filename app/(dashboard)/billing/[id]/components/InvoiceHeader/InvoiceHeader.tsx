//Components
import { Flex } from "@tremor/react";
import InvoiceLogo from "../InvoiceLogo/InvoiceLogo";

interface InvoiceHeaderProps {
  addressBank: string;
  cityBank: string;
  stateBank: string;
  phoneBank: string;
  fullName: string;
  addressCustomer: string;
  cityCustomer: string;
  stateCode: string;
  stateCustomer: string;
}

export const InvoiceHeader = ({
  addressBank,
  cityBank,
  stateBank,
  phoneBank,
  fullName,
  addressCustomer,
  cityCustomer,
  stateCode,
  stateCustomer,
}: InvoiceHeaderProps) => {
  const renderAddressBankInfo = `${addressBank} ${cityBank}, ${stateBank}`;

  return (
    <Flex className="flex-col md:flex-row">
      <Flex flexDirection="col" alignItems="start" className="md:max-w-[33%]">
        <InvoiceLogo width={16} height={16} />
        <p className="text-primary dark:text-white font-bold leading-6 tracking-wide mt-6">
          {renderAddressBankInfo}
        </p>
        <p className="text-secondary dark:text-dark-romance font-primary font-light leading-6 tracking-wide mt-2 mb-4">
          {phoneBank}
        </p>
      </Flex>
      <Flex
        flexDirection="col"
        className="items-start md:items-end mt-10 md:mt-0">
        <p className="text-primary dark:text-white font-bold leading-6 tracking-wide">
          Billed to: {fullName}
        </p>
        <p className="text-secondary dark:text-dark-romance font-primary font-light leading-6 tracking-wide md:text-right">
          {addressCustomer}
          <br /> {cityCustomer} {stateCode}
          <br /> {stateCustomer}
        </p>
      </Flex>
    </Flex>
  );
};

export default InvoiceHeader;
