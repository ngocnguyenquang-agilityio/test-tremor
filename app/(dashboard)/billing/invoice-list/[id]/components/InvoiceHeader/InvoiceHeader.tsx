//Components
import { Flex } from "@tremor/react";
import InvoiceLogo from "../InvoiceLogo/InvoiceLogo";

interface InvoiceHeaderProps {
  addressBank: string;
  cityBank: string;
  stateBank: string;
  phoneBank: string;
  fullName: string;
  addressCustomer: string;
  cityCustomer: string;
  stateCode: string;
  stateCustomer: string;
}

export const InvoiceHeader = ({
  addressBank,
  cityBank,
  stateBank,
  phoneBank,
  fullName,
  addressCustomer,
  cityCustomer,
  stateCode,
  stateCustomer,
}: InvoiceHeaderProps) => {
  const renderAddressBankInfo = `${addressBank} ${cityBank}, ${stateBank}`;

  return (
    <Flex className="flex-col md:flex-row mb-[40px] md:mb-20">
      <Flex flexDirection="col" alignItems="start" className="md:max-w-[35%]">
        <InvoiceLogo additionalClasses="w-20 h-20 md:w-10 md:h-10" />
        <p className="text-primary text-base dark:text-white font-semibold leading-6 tracking-wide mt-7">
          {renderAddressBankInfo}
        </p>
        <p className="text-secondary text-base dark:text-dark-romance font-primary font-light leading-6 tracking-wide mt-2 mb-6 md:mb-4">
          tel: {phoneBank}
        </p>
      </Flex>
      <Flex
        flexDirection="col"
        className="items-start md:items-end mt-11 md:mt-0">
        <p className="text-primary text-base dark:text-white font-semibold leading-6 tracking-wide">
          Billed to: {fullName}
        </p>
        <p className="text-secondary text-base dark:text-dark-romance font-primary font-light leading-6 tracking-wide md:text-right">
          {addressCustomer}
          <br /> {cityCustomer} {stateCode}
          <br /> {stateCustomer}
        </p>
      </Flex>
    </Flex>
  );
};

export default InvoiceHeader;
