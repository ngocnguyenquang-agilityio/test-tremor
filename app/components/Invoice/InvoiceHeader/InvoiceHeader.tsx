//Components
import { Flex } from "@tremor/react";
import InvoiceLogo from "@/components/Invoice/InvoiceLogo/InvoiceLogo";

interface BankInfo {
  address: string;
  city: string;
  state: string;
  phone: string;
}

interface Customer {
  fullName: string;
  address: string;
  city: string;
  stateCode: string;
  state: string;
}

interface InvoiceHeaderProps {
  bankInfo: BankInfo;
  customer: Customer;
}

export const InvoiceHeader = ({ bankInfo, customer }: InvoiceHeaderProps) => {
  const renderAddressBankInfo = `${bankInfo.address} ${bankInfo.city}, ${bankInfo.state}`;

  return (
    <Flex className="flex-col md:flex-row mb-[40px] md:mb-20">
      <Flex flexDirection="col" alignItems="start" className="md:max-w-[35%]">
        <InvoiceLogo additionalClasses="w-20 h-20 md:w-10 md:h-10" />
        <p className="text-primary text-base dark:text-white font-semibold leading-6 tracking-wide mt-7">
          {renderAddressBankInfo}
        </p>
        <p className="text-secondary text-base dark:text-dark-romance font-primary font-light leading-6 tracking-wide mt-2 mb-6 md:mb-4">
          tel: {bankInfo.phone}
        </p>
      </Flex>
      <Flex
        flexDirection="col"
        className="items-start md:items-end mt-11 md:mt-0">
        <p className="text-primary text-base dark:text-white font-semibold leading-6 tracking-wide">
          Billed to: {customer.fullName}
        </p>
        <p className="text-secondary text-base dark:text-dark-romance font-primary font-light leading-6 tracking-wide md:text-right max-w-full md:max-w-sm truncate">
          {customer.address}
          <br /> {customer.city} {customer.stateCode}
          <br /> {customer.state}
        </p>
      </Flex>
    </Flex>
  );
};

export default InvoiceHeader;
