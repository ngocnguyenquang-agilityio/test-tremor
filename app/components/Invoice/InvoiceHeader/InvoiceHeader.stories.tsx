// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import InvoiceHeader from "./InvoiceHeader";

const meta = {
  title: "Components/Invoice/InvoiceHeader",
  tags: ["autodocs"],
  component: InvoiceHeader,
} as Meta<typeof InvoiceHeader>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => <InvoiceHeader />,
};
