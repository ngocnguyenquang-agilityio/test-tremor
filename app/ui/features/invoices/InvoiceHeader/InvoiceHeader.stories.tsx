// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import InvoiceHeader from "./InvoiceHeader";

const InvoiceHeaderProps = {
  addressBank: "St. Independence Embankment, 050105",
  cityBank: "Bucharest",
  stateBank: "Romania",
  phoneBank: "+4 (074) 1090873",
  fullName: "John Doe",
  addressCustomer: "4006 Locust View Drive",
  cityCustomer: "San Francisco",
  stateCode: "CA",
  stateCustomer: "California",
};

const meta = {
  title: "Components/Billing/Invoices/InvoiceHeader",
  tags: ["autodocs"],
  component: InvoiceHeader,
  argTypes: {
    addressBank: { description: "Address of bank of site" },
    cityBank: { description: "City of bank of site" },
    stateBank: { description: "State of bank of site" },
    phoneBank: { description: "Phone number of site" },
    fullName: { description: "Full name of customer" },
    addressCustomer: { description: "Address of customer" },
    cityCustomer: { description: "City of customer" },
    stateCode: { description: "Code state of customer" },
    stateCustomer: { description: "State of customer" },
  },
} as Meta<typeof InvoiceHeader>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => (
    <div className="dark:bg-dark_blue text-primary rounded-xl p-6 shadow-box-icon-default dark:shadow-main-content">
      <InvoiceHeader {...InvoiceHeaderProps} />
    </div>
  ),
};
