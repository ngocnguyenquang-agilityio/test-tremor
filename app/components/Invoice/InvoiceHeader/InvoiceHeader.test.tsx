import { render } from "@testing-library/react";

// Components
import InvoiceHeader from "./InvoiceHeader";

describe("Testing InvoiceHeader component", () => {
  it("Should match snapshot", () => {
    const { container } = render(<InvoiceHeader />);
    expect(container).toMatchSnapshot();
  });
});
