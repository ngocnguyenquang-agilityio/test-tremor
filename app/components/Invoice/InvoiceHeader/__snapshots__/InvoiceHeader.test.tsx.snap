// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InvoiceHeader component Should match snapshot 1`] = `
<div>
  <div
    class="tremor-Flex-root flex w-full justify-between items-center flex-col md:flex-row"
  >
    <div
      class="tremor-Flex-root flex w-full flex-col justify-between items-start md:max-w-[33.333333%]"
    >
      <div
        class="w-16 h-16 bg-contain bg-[url('/assets/images/logo/logo-td-dark.webp')] dark:bg-[url('/assets/images/logo/logo-td-light.webp')]"
      />
      <p
        class="text-primary dark:text-white font-bold leading-[26px] tracking-[0.12px] mt-3"
      >
        St. Independence Embankment, 050105 Bucharest, Romania
      </p>
      <p
        class="text-secondary dark:text-dark-romance font-primary font-light leading-[25.6px] tracking-[0.17136px] mt-2 mb-4"
      >
        tel: +4 (074) 1090873
      </p>
    </div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-between items-start md:items-end mt-10 md:mt-0"
    >
      <p
        class="text-primary dark:text-white font-bold leading-[26px] tracking-[0.12px]"
      >
        Billed to: John Doe
      </p>
      <p
        class="text-secondary dark:text-dark-romance font-primary font-light leading-[25.6px] tracking-[0.17136px] md:text-right"
      >
        4006 Locust View Drive
        <br />
         San Francisco CA
        <br />
         California
      </p>
    </div>
  </div>
</div>
`;
